### ISSUE TITLE
kubevirt fails to launch a VMI with a PVC disk

### ISSUE THREAD

**Issue by praveen-pk** [kubevirt/kubevirt#11951]
**What happened**:
A VMI with a disk configured over NFS PVC volume fails to start. 

**What you expected to happen**:
The VMI starts and console is accessible.

**How to reproduce it (as minimally and precisely as possible)**:
I noticed this failure while using kind cluster.

Below are the steps to reproduce:

1) Create a Kind Cluster
```
[ $(uname -m) = x86_64 ] && curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.22.0/kind-linux-amd64


# Cluster with 1 control-plane and one node
$ cat kind_cluster.yaml
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
- role: worker


kind create cluster --name test --config kind_cluster.yaml
```
2) Install Kubevirt Components
```
# Point at latest release
export RELEASE=$(curl https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirt/stable.txt)

# Deploy the KubeVirt operator
kubectl apply -f https://github.com/kubevirt/kubevirt/releases/download/${RELEASE}/kubevirt-operator.yaml

# Create the KubeVirt CR (instance deployment request) which triggers the actual installation
kubectl apply -f https://github.com/kubevirt/kubevirt/releases/download/${RELEASE}/kubevirt-cr.yaml

# Download virtctl
wget https://github.com/kubevirt/kubevirt/releases/download/${RELEASE}/virtctl-${RELEASE}-linux-amd64


# wait until all KubeVirt components are up
kubectl -n kubevirt wait kv kubevirt --for condition=Available


# Kind might need below command to upload the guest image before starting
kind load docker-image quay.io/kubevirt/cirros-container-disk-demo:latest --name test
```

3) Setup NFS based PVC
```
$ cat nfs_pv.yaml
apiVersion: v1
kind: PersistentVolume
metadata:
  name: nfs
spec:
  volumeMode: Filesystem
  storageClassName: nfs
  capacity:
    storage: 2048Mi
  accessModes:
    - ReadWriteMany
  nfs:
    server: 192.168.68.128
    path: "/SHARED/NFS"
  mountOptions:
    - nfsvers=4.2
    - hard

$ kubectl apply -f nfs_pv.yaml

$ kubectl get pv
NAME   CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM            STORAGECLASS   VOLUMEATTRIBUTESCLASS   REASON   AGE
nfs    2Gi        RWX            Retain           Bound    default/nfspvc   nfs            <unset>                          22h

$ cat nfs_pvc.yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: nfspvc
spec:
  accessModes:
    - ReadWriteMany
  storageClassName: nfs
  resources:
    requests:
      storage: 1024Mi
  volumeName: nfs

$ kubectl apply -f nfs_pvc.yaml 

$ kubectl get pvc
NAME     STATUS   VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS   VOLUMEATTRIBUTESCLASS   AGE
nfspvc   Bound    nfs      2Gi        RWX            nfs            <unset>                 22h

``` 

4) start a VM
```
$ cat nfsvm_disk.yaml
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  name: nfsvm
spec:
  running: false
  template:
    metadata:
      labels:
        kubevirt.io/size: small
        kubevirt.io/domain: nfsvm
    spec:
      domain:
        devices:
          disks:
            - name: containerdisk
              disk:
                bus: virtio
            - name: cloudinitdisk
              disk:
                bus: virtio
            - name: nfspvcdisk
              disk: {}
          interfaces:
          - name: default
            masquerade: {}
        resources:
          requests:
            memory: 64M
      networks:
      - name: default
        pod: {}
      volumes:
        - name: containerdisk
          containerDisk:
            image: quay.io/kubevirt/cirros-container-disk-demo
        - name: cloudinitdisk
          cloudInitNoCloud:
            userDataBase64: SGkuXG4=
        - name: nfspvcdisk
          persistentVolumeClaim:
            claimName: nfspvc

$ kubectl apply -f nfsvm_disk.yaml
virtualmachine.kubevirt.io/nfsvm created

$ kubectl get vms
NAME    AGE   STATUS     READY
nfsvm   12s   Starting   False

$ virtctl start nfsvm
VM nfsvm was scheduled to start

$ kubectl get pods
NAME                        READY   STATUS            RESTARTS   AGE
virt-launcher-nfsvm-6qn24   0/3     PodInitializing   0          31s
$ kubectl get pods
NAME                        READY   STATUS   RESTARTS   AGE
virt-launcher-nfsvm-6qn24   0/3     Error    0          4m54s

```

**Additional context**:
With above configure NFS based PVC I confirmed I can access files in PVC from a pod with below configuration:
```
$ cat nfs_pvc_pod.yaml
kind: Pod
apiVersion: v1
metadata:
  name: nfs-pvc-pod
spec:
   volumes:
    - name: nginx-pv-storage
      persistentVolumeClaim:
        claimName: nfspvc
   containers:
    - name: nginx
      image: nginx
      ports:
        - containerPort: 80
          name: "nginx-server"
      volumeMounts:
        - mountPath: "/usr/share/nginx/html"
          name: nginx-pv-storage
```
This tells me the PVC configuration is correct.

***Logs***
### Controller
Below are the logs from controller, showing the Backoff flow, but nothing related to underlying failure:

```
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:904","timestamp":"2024-05-21T17:16:04.725462Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Dispatching delete event for vmi default/nfsvm with phase Failed","name":"nfsvm","namespace":"default","pos":"vm.go:1496","timestamp":"2024-05-21T17:16:04.739888Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"nfsvm","namespace":"default","pos":"vm.go:386","reason":"the server rejected our request due to an error in our request","timestamp":"2024-05-21T17:16:04.775541Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/nfsvm","pos":"vm.go:280","reason":"the server rejected our request due to an error in our request","timestamp":"2024-05-21T17:16:04.775597Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:904","timestamp":"2024-05-21T17:16:04.775683Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:904","timestamp":"2024-05-21T17:16:04.806897Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"nfsvm","namespace":"default","pos":"vm.go:386","reason":"virtualmachineinstances.kubevirt.io \"nfsvm\" not found","timestamp":"2024-05-21T17:16:04.809420Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/nfsvm","pos":"vm.go:280","reason":"virtualmachineinstances.kubevirt.io \"nfsvm\" not found","timestamp":"2024-05-21T17:16:04.809471Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Delaying start of VM Starting VM with 'runStrategy: Always' due to start failure backoff. Waiting 10 more seconds before starting.","name":"nfsvm","namespace":"default","pos":"vm.go:923","timestamp":"2024-05-21T17:16:04.809551Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Delaying start of VM Starting VM with 'runStrategy: Always' due to start failure backoff. Waiting 10 more seconds before starting.","name":"nfsvm","namespace":"default","pos":"vm.go:923","timestamp":"2024-05-21T17:16:04.822946Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"nfsvm","namespace":"default","pos":"vm.go:386","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:16:04.829029Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/nfsvm","pos":"vm.go:280","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:16:04.829084Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Delaying start of VM Starting VM with 'runStrategy: Always' due to start failure backoff. Waiting 10 more seconds before starting.","name":"nfsvm","namespace":"default","pos":"vm.go:923","timestamp":"2024-05-21T17:16:04.829164Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Delaying start of VM Starting VM with 'runStrategy: Always' due to start failure backoff. Waiting 10 more seconds before starting.","name":"nfsvm","namespace":"default","pos":"vm.go:923","timestamp":"2024-05-21T17:16:04.834627Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Starting VM due to runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:928","timestamp":"2024-05-21T17:16:14.835423Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
W0521 17:16:14.877747       1 warnings.go:70] metadata.finalizers: "foregroundDeleteVirtualMachine": prefer a domain-qualified finalizer name to avoid accidental conflicts with other finalizer writers
{"component":"virt-controller","kind":"","level":"info","msg":"Started VM by creating the new virtual machine instance nfsvm","name":"nfsvm","namespace":"default","pos":"vm.go:1192","timestamp":"2024-05-21T17:16:14.878356Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"nfsvm","namespace":"default","pos":"vm.go:386","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:16:14.912174Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/nfsvm","pos":"vm.go:280","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:16:14.912270Z"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"nfsvm","namespace":"default","pos":"vm.go:386","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:16:14.943591Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/nfsvm","pos":"vm.go:280","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:16:14.943639Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:904","timestamp":"2024-05-21T17:22:07.687614Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Dispatching delete event for vmi default/nfsvm with phase Failed","name":"nfsvm","namespace":"default","pos":"vm.go:1496","timestamp":"2024-05-21T17:22:07.696112Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:904","timestamp":"2024-05-21T17:22:07.716975Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"nfsvm","namespace":"default","pos":"vm.go:386","reason":"the server rejected our request due to an error in our request","timestamp":"2024-05-21T17:22:07.721470Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/nfsvm","pos":"vm.go:280","reason":"the server rejected our request due to an error in our request","timestamp":"2024-05-21T17:22:07.721500Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:904","timestamp":"2024-05-21T17:22:07.721544Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"reenqueuing VirtualMachineInstance default/nfsvm","pos":"vmi.go:322","reason":"Operation cannot be fulfilled on virtualmachineinstances.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:22:07.723781Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:904","timestamp":"2024-05-21T17:22:07.727539Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Delaying start of VM Starting VM with 'runStrategy: Always' due to start failure backoff. Waiting 42 more seconds before starting.","name":"nfsvm","namespace":"default","pos":"vm.go:923","timestamp":"2024-05-21T17:22:07.733009Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Delaying start of VM Starting VM with 'runStrategy: Always' due to start failure backoff. Waiting 42 more seconds before starting.","name":"nfsvm","namespace":"default","pos":"vm.go:923","timestamp":"2024-05-21T17:22:07.741735Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Starting VM due to runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:928","timestamp":"2024-05-21T17:22:49.733738Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
W0521 17:22:49.769155       1 warnings.go:70] metadata.finalizers: "foregroundDeleteVirtualMachine": prefer a domain-qualified finalizer name to avoid accidental conflicts with other finalizer writers
{"component":"virt-controller","kind":"","level":"info","msg":"Started VM by creating the new virtual machine instance nfsvm","name":"nfsvm","namespace":"default","pos":"vm.go:1192","timestamp":"2024-05-21T17:22:49.769742Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"nfsvm","namespace":"default","pos":"vm.go:386","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:22:49.803313Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/nfsvm","pos":"vm.go:280","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:22:49.803423Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:904","timestamp":"2024-05-21T17:28:32.735676Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Dispatching delete event for vmi default/nfsvm with phase Failed","name":"nfsvm","namespace":"default","pos":"vm.go:1496","timestamp":"2024-05-21T17:28:32.747385Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:904","timestamp":"2024-05-21T17:28:32.863980Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"nfsvm","namespace":"default","pos":"vm.go:386","reason":"the server rejected our request due to an error in our request","timestamp":"2024-05-21T17:28:32.871712Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/nfsvm","pos":"vm.go:280","reason":"the server rejected our request due to an error in our request","timestamp":"2024-05-21T17:28:32.871806Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:904","timestamp":"2024-05-21T17:28:32.871937Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:904","timestamp":"2024-05-21T17:28:32.877315Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"reenqueuing VirtualMachineInstance default/nfsvm","pos":"vmi.go:322","reason":"Operation cannot be fulfilled on virtualmachineinstances.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:28:32.877364Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Delaying start of VM Starting VM with 'runStrategy: Always' due to start failure backoff. Waiting 102 more seconds before starting.","name":"nfsvm","namespace":"default","pos":"vm.go:923","timestamp":"2024-05-21T17:28:32.906674Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Delaying start of VM Starting VM with 'runStrategy: Always' due to start failure backoff. Waiting 102 more seconds before starting.","name":"nfsvm","namespace":"default","pos":"vm.go:923","timestamp":"2024-05-21T17:28:32.930015Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"info","msg":"Starting VM due to runStrategy: Always","name":"nfsvm","namespace":"default","pos":"vm.go:928","timestamp":"2024-05-21T17:30:14.907086Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
W0521 17:30:14.936738       1 warnings.go:70] metadata.finalizers: "foregroundDeleteVirtualMachine": prefer a domain-qualified finalizer name to avoid accidental conflicts with other finalizer writers
{"component":"virt-controller","kind":"","level":"info","msg":"Started VM by creating the new virtual machine instance nfsvm","name":"nfsvm","namespace":"default","pos":"vm.go:1192","timestamp":"2024-05-21T17:30:14.936904Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"nfsvm","namespace":"default","pos":"vm.go:386","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:30:14.948707Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/nfsvm","pos":"vm.go:280","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:30:14.948770Z"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"nfsvm","namespace":"default","pos":"vm.go:386","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:30:14.961671Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/nfsvm","pos":"vm.go:280","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:30:14.961700Z"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"nfsvm","namespace":"default","pos":"vm.go:386","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:30:14.980767Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/nfsvm","pos":"vm.go:280","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:30:14.980797Z"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"nfsvm","namespace":"default","pos":"vm.go:386","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:30:14.992155Z","uid":"2e912a60-42d7-4fe8-bc27-4d85e59e21af"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/nfsvm","pos":"vm.go:280","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"nfsvm\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2024-05-21T17:30:14.992184Z"}
```

### Launcher
virt-launcher reaches ready state, but does not progress any further:

```
$ kubectl  logs -f -c compute $(kubectl get pods -o name| grep ${VM_NAME})
{"component":"virt-launcher","level":"info","msg":"Collected all requested hook sidecar sockets","pos":"manager.go:88","timestamp":"2024-05-21T17:11:08.070654Z"}
{"component":"virt-launcher","level":"info","msg":"Sorted all collected sidecar sockets per hook point based on their priority and name: map[]","pos":"manager.go:91","timestamp":"2024-05-21T17:11:08.071487Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon: qemu+unix:///session?socket=/var/run/libvirt/virtqemud-sock","pos":"libvirt.go:565","timestamp":"2024-05-21T17:11:08.072186Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon failed: virError(Code=38, Domain=7, Message='Failed to connect socket to '/var/run/libvirt/virtqemud-sock': No such file or directory')","pos":"libvirt.go:573","timestamp":"2024-05-21T17:11:08.073200Z"}
{"component":"virt-launcher","level":"info","msg":"libvirt version: 9.5.0, package: 6.el9 (builder@centos.org, 2023-08-25-08:53:56, )","subcomponent":"libvirt","thread":"47","timestamp":"2024-05-21T17:11:08.106000Z"}
{"component":"virt-launcher","level":"info","msg":"hostname: nfsvm","subcomponent":"libvirt","thread":"47","timestamp":"2024-05-21T17:11:08.106000Z"}
{"component":"virt-launcher","level":"error","msg":"Unable to open /dev/kvm: Permission denied","pos":"virHostCPUGetCPUID:1422","subcomponent":"libvirt","thread":"47","timestamp":"2024-05-21T17:11:08.106000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: Unable to get session bus connection: Cannot spawn a message bus without a machine-id: Unable to load /var/lib/dbus/machine-id or /etc/machine-id: Failed to open file “/var/lib/dbus/machine-id”: No such file or directory","pos":"virGDBusGetSessionBus:126","subcomponent":"libvirt","thread":"47","timestamp":"2024-05-21T17:11:08.107000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: Unable to get system bus connection: Could not connect: No such file or directory","pos":"virGDBusGetSystemBus:99","subcomponent":"libvirt","thread":"47","timestamp":"2024-05-21T17:11:08.107000Z"}
{"component":"virt-launcher","level":"info","msg":"Connected to libvirt daemon","pos":"libvirt.go:581","timestamp":"2024-05-21T17:11:08.577021Z"}
{"component":"virt-launcher","level":"info","msg":"Registered libvirt event notify callback","pos":"client.go:563","timestamp":"2024-05-21T17:11:08.586593Z"}
{"component":"virt-launcher","level":"info","msg":"Marked as ready","pos":"virt-launcher.go:75","timestamp":"2024-05-21T17:11:08.587328Z"}
```

**Environment**:
- KubeVirt version (use `virtctl version`):  ```
Client Version: version.Info{GitVersion:"v1.2.0", GitCommit:"f26e45d99ac35743fc33d6a121b629e9a9af6b63", GitTreeState:"clean", BuildDate:"2024-03-05T20:34:24Z", GoVersion:"go1.21.5 X:nocoverageredesign", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{GitVersion:"v1.2.0", GitCommit:"f26e45d99ac35743fc33d6a121b629e9a9af6b63", GitTreeState:"clean", BuildDate:"2024-03-05T21:32:21Z", GoVersion:"go1.21.5 X:nocoverageredesign", Compiler:"gc", Platform:"linux/amd64"} ```

- Kubernetes version (use `kubectl version`): ``` Client Version: v1.29.5
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.29.2```

- VM or VMI specifications: Provided Above
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): ```$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy ``` 
- Kernel (e.g. `uname -a`): ```Linux ubuntu-dom0 5.15.0-107-generic #117-Ubuntu SMP Fri Apr 26 12:26:49 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux```
- Install tools: N/A
- Others: N/A

**Comment by awels** [comment]
A few questions, can you check if your node is presenting the kvm device? `kubectl get nodes <name> -o=jsonpath={.status.allocatable}` if it is presenting the kvm devices there should be something like this in the output `"devices.kubevirt.io/kvm":"1k"`.

Second I don't see you populating the PVC with a disk image at all, so are you sure there is a proper disk image on the PVC for kubevirt to use? If not I would suggest using CDI to populate the disk image that you are attempting to use.

**Comment by praveen-pk** [comment]
Thanks for the prompt reponse!

> A few questions, can you check if your node is presenting the kvm device? `kubectl get nodes <name> -o=jsonpath={.status.allocatable}` if it is presenting the kvm devices there should be something like this in the output `"devices.kubevirt.io/kvm":"1k"`.


I see this is configured correctly:
```
$ kubectl get nodes como-worker -o=jsonpath={.status.allocatable}
{"cpu":"8","devices.kubevirt.io/kvm":"1k","devices.kubevirt.io/tun":"1k","devices.kubevirt.io/vhost-net":"1k","ephemeral-storage":"238169856Ki","hugepages-1Gi":"0","hugepages-2Mi":"0","memory":"32683504Ki","pods":"110"}
```

> 
> Second I don't see you populating the PVC with a disk image at all, so are you sure there is a proper disk image on the PVC for kubevirt to use? If not I would suggest using CDI to populate the disk image that you are attempting to use.

Below are the files I am exporting via PVC:
```
$ ls /SHARED/NFS/
disk3.img  disk3.raw  index.html  nfs-test.txt
```

Above disk images were manually created with `dd`. As this is a data disk, I didn't put any data in them. Should these disk images be of any specific format?

**Comment by awels** [comment]
So for filesystem PVCs there should be a single disk.img file in the root of the volume. KubeVirt tries to find that image file. If you want to create just a blank disk, I highly suggest you use CDI to create a DataVolume with a blank source. https://github.com/kubevirt/containerized-data-importer/blob/main/doc/datavolumes.md#blank-data-volume

**Comment by praveen-pk** [comment]
I am actually trying to attach a persistent disk from a PVC to a guest. I am using a blank disk just for initial testing. 

Following above recommendation, I created a DataVolume with blank source and was able to confirm the guest can see the new volume. But this isn't my true goal. 

Following, https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#persistentvolumeclaim, I tried to expose the PVC as a `lun` with :
```
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  name: nfsvm.lun
spec:
  running: false
  template:
    metadata:
      labels:
        kubevirt.io/size: small
        kubevirt.io/domain: nfsvm.lun
    spec:
      domain:
        devices:
          disks:
            - name: containerdisk
              disk:
                bus: virtio
            - name: cloudinitdisk
              disk:
                bus: virtio
            - name: nfspvcdisk
              lun: {}
          interfaces:
          - name: default
            masquerade: {}
        resources:
          requests:
            memory: 64M
      networks:
      - name: default
        pod: {}
      volumes:
        - name: containerdisk
          containerDisk:
            image: quay.io/kubevirt/cirros-container-disk-demo
        - name: cloudinitdisk
          cloudInitNoCloud:
            userDataBase64: SGkuXG4=
        - name: nfspvcdisk
          persistentVolumeClaim:
            claimName: nfspvc
```

I also tried the following  https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#disk:

```
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  name: nfsvm.disk
spec:
  running: false
  template:
    metadata:
      labels:
        kubevirt.io/size: small
        kubevirt.io/domain: nfsvm.disk
    spec:
      domain:
        devices:
          disks:
            - name: containerdisk
              disk:
                bus: virtio
            - name: cloudinitdisk
              disk:
                bus: virtio
            - name: nfspvcdisk
              disk: {}
          interfaces:
          - name: default
            masquerade: {}
        resources:
          requests:
            memory: 64M
      networks:
      - name: default
        pod: {}
      volumes:
        - name: containerdisk
          containerDisk:
            image: quay.io/kubevirt/cirros-container-disk-demo
        - name: cloudinitdisk
          cloudInitNoCloud:
            userDataBase64: SGkuXG4=
        - name: nfspvcdisk
          persistentVolumeClaim:
            claimName: nfspvc
```

With both these configs, I am seeing the same behavior where `virt-laucher` reaches `Marked as ready` and never progresses any further.

Now, I added `disk.img` to NFS share: 

```
$ ls -lh /SHARED/NFS/
total 101M
-rwxrwxrwx. 1 root root 100M May 22 12:32 disk.img
-rw-r--r--. 1 root root  296 Apr 23 14:28 index.html
-rw-r--r--. 1 root root    0 Mar 13 18:10 nfs-test.txt
```
This does not seem to have any impact on the failure. Any thoughts on what else I could be missing?

**Comment by praveen-pk** [comment]
@awels  I finally got the VM to boot with a PVC based on NFS.

 As you pointed out I had to create `disk.img` file at the root of NFS. Another configuration I missed was the owning uid of the disk.img. Seems like I need to set the owning uid of `disk.img` to `107` to allow `qemu` to access the file. This is documented in https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#persistentvolumeclaim. 

It would have been great if `virt-launcher` reported some permissions or other error, when it couldn't run the guest.
